File: src/main/java/com/mellivorines/generator/entity/GenDatasource.java
package com.mellivorines.generator.entity;


import org.babyfish.jimmer.sql.Column;
import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.GeneratedValue;
import org.babyfish.jimmer.sql.GenerationType;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.Table;
import org.jetbrains.annotations.Nullable;

@Entity
@Table(name = "gen_datasource")
public interface GenDatasource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    int id();


    @Nullable
    @Column(name = "db_type")
    String dbType();

    @Nullable
    @Column(name = "conn_name")
    String connName();

    @Nullable
    @Column(name = "conn_url")
    String connUrl();

    @Nullable
    @Column(name = "username")
    String username();

    @Nullable
    @Column(name = "password")
    String password();

}

File: src/main/kotlin/com/mellivorines/generator/entity/GenDatasource.kt
package com.mellivorines.generator.entity

import org.babyfish.jimmer.sql.Column
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.GeneratedValue
import org.babyfish.jimmer.sql.GenerationType
import org.babyfish.jimmer.sql.Id
import org.babyfish.jimmer.sql.Table

@Entity
@Table(name = "gen_datasource")
interface GenDatasource {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long

    @Column(name = "db_type")
    val dbType: String?

    @Column(name = "conn_name")
    val connName: String?

    @Column(name = "conn_url")
    val connUrl: String?

    @Column(name = "username")
    val username: String?

    @Column(name = "password")
    val password: String?
}